### ISSUE TITLE
[Bug] RandomCrop shifts bounding box coordinates along the wrong axes (h/w swapped)

### ISSUE THREAD

**Issue by patminn107** [keras-team/keras#23518]
**System information**
- Keras version: 3.13.2
- Backend: tensorflow
- Environment: Google Colab (Python 3.13)

**Describe the bug**

`keras.layers.RandomCrop.transform_bounding_boxes` subtracts the **vertical** crop
offset `h_start` from the **x** coordinates and the **horizontal** offset `w_start`
from the **y** coordinates.

In `xyxy` format (`[x1, y1, x2, y2]`), x coordinates must be shifted by `w_start`
and y coordinates by `h_start`. The offsets are applied to the wrong axes, which
silently corrupts detection labels whenever `h_start != w_start` — i.e. virtually
every training step, since both offsets are drawn independently at random.

`CenterCrop` performs the equivalent shift correctly (x↔w, y↔h), which confirms
this is a bug and not a naming convention.

- Batched branch (rank-3 boxes):
  https://github.com/keras-team/keras/blob/master/keras/src/layers/preprocessing/image_preprocessing/random_crop.py#L237-L240
- Unbatched branch (rank-2 boxes):
  https://github.com/keras-team/keras/blob/master/keras/src/layers/preprocessing/image_preprocessing/random_crop.py#L247-L250

**Standalone code to reproduce the issue**

```python
import numpy as np
import keras

layer = keras.layers.RandomCrop(2, 4, bounding_box_format="xyxy")

out = layer.transform_bounding_boxes(
    {"boxes": np.array([[0.0, 0.0, 8.0, 8.0]]), "labels": np.array([0])},
    (3, 1),   # h_start=3, w_start=1: the crop starts at row 3, column 1
    training=True,
)
print(out["boxes"])
```
Expected behavior

The box is shifted horizontally by w_start=1 and vertically by h_start=3:

[[0. 0. 7. 5.]]   # x1=max(0-1,0), y1=max(0-3,0), x2=max(8-1,0), y2=max(8-3,0)
Actual behavior

tf.Tensor([[0. 0. 5. 7.]], shape=(1, 4), dtype=float32)   # x shifted by h_start, y shifted by w_start (swapped)
Impact

Object-detection models using RandomCrop with bounding boxes train on silently corrupted labels — no error, no warning. The existing test (test_dict_input in random_crop_test.py) only checks the output shape, never the coordinate values, so the bug ships unnoticed.

Related observations (possible follow-ups, not part of this report)

Only a lower bound (maximum(..., 0)) is applied; coordinates are not clipped to the crop size as CenterCrop does.
With relative formats (rel_xyxy, ...), the initial conversion to xyxy uses the crop dimensions (self.height/self.width) instead of the input image dimensions.
Would you like to help us fix it? Yes — happy to open a PR with the fix and value-based tests for both branches.

**Comment by patminn107** [comment]
Opened a PR with the fix: https://github.com/keras-team/keras/pull/23519

It swaps the h/w offsets in both branches of `transform_bounding_boxes`, clips
the shifted boxes to the crop size, and adds regression tests with explicit
expected values for batched and unbatched boxes. Tests pass on the tensorflow
backend (10/10).

Happy to adjust anything based on review feedback.

**Comment by indrajeetapache** [comment]
@mrinalghoshh  can i work on this issue ?

**Comment by patminn107** [comment]
Hi @indrajeetapache, this issue is already being fixed in PR #23519, which
I opened earlier today. The offset swap + regression tests are up for review.
If you'd like to help, reviewing that PR would be the most useful, thanks!
